src/app/data-planes/views/DataPlaneInboundSummaryStatsView.vue

```diff
@@ -30,15 +30,41 @@
           />
 
           <LoadingBlock v-else-if="data === undefined" />
-
           <CodeBlock
             v-else
             language="json"
-            :code="(() => data.raw.split('\n')
-              .filter((item) => item.includes(`.${route.params.service}.`))
-              .map((item) => item.replace(`${route.params.service}.`, ''))
-              .join('\n')
-            )()"
+            :code="(() => {
+              if(props.gateway) {
+                const parts = route.params.service.split(':')
+                return data.raw.split('\n')
+                  .filter((item) => item.includes(`.${parts[0]}.`))
+                  .filter((item) => {
+                    return !item.startsWith('listener.') || item.includes(`_${parts[1]}.`)
+                  })
+                  .filter((item) => {
+                    return !item.includes('.rds.') || item.includes(`_${parts[1]}_*.`)
+                  })
+                  .map((item) => item.replace(`${parts[0]}.`, ''))
+                  .map((item) => item.replace(`_${parts[1]}.`, '.'))
+                  .map((item) => {
+                    if(item.includes('.rds.')) {
+                      const tmp = item.split('.')
+                      tmp.splice(2, 1)
+                      return tmp.join('.')
+                    } else {
+                      return item;
+                    }
+                  })
+                  .join('\n')
+
+              } else {
+                const name = `localhost_${route.params.service.split(':')[1]}`
+                return data.raw.split('\n')
+                  .filter((item) => item.includes(`.${name}.`))
+                  .map((item) => item.replace(`${name}.`, ''))
+                  .join('\n')
+              }
+            })()"
```

Comment:
**Suggestion**: Can we introduce a function for this somewhere? There is a ton of stuff going on here that shouldn’t be in the template here.

Reply:
Yes definitely, was gonna do something in a follow up, let me make this a little better here first.

Reply:
I've moved this into a function for the minute, it will probably go into a `/connections/data/index.ts` file later on.